### ISSUE TITLE
How to build javascript from source?

### ISSUE THREAD

**Issue by mehrdaad** [nextcloud/server#7146]
I have modified a list of js files, is there a need to build the project?  
If so, what is the proper command?

**Comment by juliushaertl** [comment]
For the server part there is no build process involved. Just change the files.

**Comment by mehrdaad** [comment]
@juliushaertl this also includes js files?

**Comment by mehrdaad** [comment]
@juliushaertl also, how to require a library located in ``core/vendor`` directory.

**Comment by juliushaertl** [comment]
To include other JS files you need to add them by calling `OC_Util::addVendorScript` in your app. For the core scripts this is done here: https://github.com/nextcloud/server/blob/dfd8125aeb4a95df20041890dcffd50ba2592fee/lib/private/legacy/template.php#L111

Please also note that GitHub is only for issue tracking. If you have any further questions on development please check the forum at https://help.nextcloud.com/c/dev or join the #nextcloud-dev irc channel on freenode.

**Comment by mehrdaad** [comment]
@juliushaertl hey, just need a clarification on this issue. Not sure to ask it here or open a discussion in irc channel.  
Any ways,  I checked ``server/lib/private/legacy/template.php`` as you mentioned.  
https://github.com/nextcloud/server/blob/dfd8125aeb4a95df20041890dcffd50ba2592fee/lib/private/legacy/template.php#L128  
Apparently, added dependencies is read from ``core.json`` and each one is added as a dependency.  
I've added another lib here, so I expect that lib should be added as a lib without further effort, but that doesn't.  
Is there another step to build js?  
Any hints?
